src/librustc_middle/mir/interpret/error.rs

```diff
@@ -356,6 +354,8 @@ pub enum UndefinedBehaviorInfo {
     InvalidBool(u8),
     /// Using a non-character `u32` as character.
     InvalidChar(u32),
+    /// An enum discriminant was set to a value which was outside the range of valid values.
+    InvalidDiscriminant(ScalarMaybeUndef),
```

Comment:
why was this moved?

Reply:
I moved it to the other `Invalid*` variants. Invalid Boolean, invalid character, invalid discriminant -- all seem pretty related to me.

Reply:
"Invalid int ptr usage" doesn't really fit, as here it's not that the value itself is fundamentally wrong but that it gets used in the wrong way... maybe we should rename that variant?

`DanglingIntPtr` would be a possible name.

Reply:
@oli-obk like so?